test/boost/sstable_set_test.cc

```diff
@@ -167,3 +172,27 @@ SEASTAR_TEST_CASE(test_partitioned_sstable_set_bytes_on_disk) {
         BOOST_REQUIRE_EQUAL(sst_set->bytes_on_disk(), ss1->bytes_on_disk() + ss2->bytes_on_disk());
     });
 }
+
+SEASTAR_TEST_CASE(test_tablet_sstable_set_copy_ctor) {
+    // enable tablets, to get access to tablet_storage_group_manager
+    cql_test_config cfg;
+    cfg.db_config->enable_tablets(true);
+    cfg.initial_tablets = 2;
+
+    return do_with_cql_env_thread([&](cql_test_env& env) {
+        env.execute_cql("CREATE KEYSPACE test_tablet_sstable_set_copy_ctor"
+                " WITH REPLICATION = {'class' : 'NetworkTopologyStrategy', 'replication_factor' : 1};").get();
+        env.execute_cql("CREATE TABLE test_tablet_sstable_set_copy_ctor.test (pk int PRIMARY KEY);").get();
+        for (int i = 0; i < 10; i++) {
+            env.execute_cql(fmt::format("INSERT INTO test_tablet_sstable_set_copy_ctor.test (pk) VALUES ({})", i)).get();
+        }
+        auto& cf = env.local_db().find_column_family("test_tablet_sstable_set_copy_ctor", "test");
+        auto& sgm = column_family_test::get_storage_group_manager(cf);
+        sgm->split_all_storage_groups().get();
+
+        replica::tablet_sstable_set tablet_sstable_set(cf.schema(), *sgm.get(), replica::tablet_map(2));
+        auto tablet_sstable_set_copy = tablet_sstable_set;
+        BOOST_REQUIRE(tablet_sstable_set_copy == tablet_sstable_set);
+
```

Comment:
could we test it through sstable_set interface? so tablet_sstable_set doesn't have to be directly exposed.

1) create tablet set through make_tablet_sstable_set().
2) copy the set, so invokes sstable_set_impl::clone(), which calls tablet_sstable_set's copy ctor
3) then perform an operation on cloned set, which requires _sstable_set_ids to be correct. for example, sstable_set::select(const dht::partition_range&) or all().

Reply:
Thank you, this worked. I have updated the code.